### PR TITLE
Readme cleanup

### PR DESCRIPTION
**Pull request by jmklix**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,15 @@
...
 __Jump To:__
 
 * [Installation](#installation)
 * [Samples](./samples)
 * [Getting Help](#getting-help)
 * [FAQ](./documents/FAQ.md)
-* [Giving Feedback and Contributions](#giving-feedback-and-contributions)
+* [API Docs](https://aws.github.io/aws-iot-device-sdk-cpp-v2/)
 * [MQTT5 User Guide](./documents/MQTT5_Userguide.md)
-* [Secure Tunnel User Guide](./documents/Secure_Tunnel_Userguide.md)
```
Comment (xiazhvera) [inline README.md:26]: Would we like to keep the link to Secure Tunnel User guide?

Comment (sbSteveK) [reply README.md:26]: +1. Alternately, we could move the secure tunnel guide elsewhere but we should have a link somewhere to it.

File: README.md [hunk 2]
```diff
@@ -54,6 +38,11 @@ cd ..
 # If working with multiple SDKs, using a SDK-specific name is helpful.
```
Comment (sbSteveK) [inline README.md:38]: Not part of this PR but this line feels a little unclear and maybe unnecessary.

File: README.md [hunk 3]
```diff
@@ -64,25 +53,11 @@ cmake --build . --target install
 ```
 
 #### Windows
-
 ``` sh
-# Create a workspace directory to hold all the SDK files
```
[the remark this answers is not in the document]
Comment (sbSteveK) [reply README.md:69]: we probably don't need to go so far as instructing someone to create a workspace directory though. I think the use of the word "workspace" could cause confusion between this directory and the "absolute path to sdk-workspace" since I don't believe they're both pointing to the same directory.

File: README.md (hunk 2 above)
Comment (sbSteveK) [reply README.md:38]: I think "git clone --recursive" makes is unnecessary to go into the folder and run "git submodule update --init --recursive" so we can probably cut out lines 49 - 52. Unless there's some reason why the submodules wouldn't be properly updated with the initial "git clone --recursive" command.

File: README.md [hunk 4]
```diff
@@ -32,10 +32,8 @@ cd sdk-workspace
 git clone --recursive https://github.com/aws/aws-iot-device-sdk-cpp-v2.git
 # Ensure all submodules are properly updated
 cd aws-iot-device-sdk-cpp-v2
```
Comment (sbSteveK) [inline README.md:34]: no need to change into this directory and change back out of this directory.

**Review by sbSteveK — APPROVED** [review]
cd into dir and cd .. ship

**Review by xiazhvera — APPROVED** [review]
The Codebuild was failed because of ubuntu-toolchain. Rerun the CI should fix it.
